lib/_http_outgoing.js

```diff
@@ -487,7 +492,7 @@ OutgoingMessage.prototype.end = function(data, encoding, callback) {
     encoding = null;
   }
 
-  if (data && typeof data !== 'string' && !(data instanceof Buffer)) {
+  if (data && !(typeof data === 'string' || data instanceof Buffer)) {
```

Comment:
I'm not sure if it's worth optimizing for the uncommon case ('bug case' really) but I'll leave that at your discretion.  I do think it's a little harder to read now, though.